fe/fe-core/src/main/java/com/starrocks/sql/ast/ColumnDef.java

```diff
@@ -347,6 +352,26 @@ public void analyze(boolean isOlap) throws AnalysisException {
                 throw new AnalysisException(
                         String.format("Invalid aggregate function '%s' for '%s'", aggregateType, name));
             }
+            // check agg_state_desc
+            if (aggregateType == AggregateType.AGG_STATE_UNION) {
+                if (aggStateDesc == null) {
+                    throw new AnalysisException(
+                            String.format("Invalid aggregate function '%s' for '%s'", aggregateType, name));
+                }
+                // Ensure agg_state_desc is compatible with type
+                AggregateFunction aggFunc = aggStateDesc.getAggregateFunction();
+                if (aggFunc == null) {
+                    throw new AnalysisException(
+                            String.format("Invalid aggregate function '%s' for '%s': aggregate function is not found",
+                                    aggregateType, name));
+                }
+                if (!aggFunc.getIntermediateTypeOrReturnType().isFullyCompatible(type)) {
+                    throw new AnalysisException(
```

Comment:
SemanticException

Reply:
`AnalysisException` is fine here because this is in `analyze` phase.